server/src/main/java/org/opensearch/search/resourcetypes/CPU.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.resourcetypes;
+
+import org.opensearch.tasks.Task;
+
+/**
+ * Represents the CPU resource type for tasks.
+ * This class provides methods to retrieve CPU usage and convert threshold percentages to long values.
+ */
+public class CPU extends ResourceType {
```

Comment:
These resource types seems little out of place for me in context of keeping `ResourceType` generic. Since here we are tying the idea to a `Task`. Now whoever want to use this class has unnecessary baggage of task related methods.

I am more in favor of existing enum `ResourceType`. We can create decorators on this enum values and use for our use case. 

What do you think @jainankitk ?

Reply:
Resolving this considering we agreed on keeping this enum untouched, more on this [here](https://github.com/opensearch-project/OpenSearch/pull/13897#issuecomment-2253101817)